c/tskit/tables.c

```diff
@@ -71,8 +70,7 @@ check_table_overflow(tsk_size_t current_size, tsk_size_t additional_rows)
 static bool
 check_offset_overflow(tsk_size_t current_size, tsk_size_t additional_elements)
 {
-    uint64_t new_size = (uint64_t) current_size + (uint64_t) additional_elements;
```

Comment:
Let's not change the logic here yet, we know this works. Your version should work, but I can see some corner cases where it might fall over, so let's not change too many things at once.

Reply:
Ok, I'll save this patch for the 64bit branch. I don't think we can guarantee that a 128bit type is available so that branch will need to use something like this.

Reply:
It's pretty academic at that point though, no conceivable computer could overflow a 64 bit number.